Source/CarthageKit/Project.swift

```diff
@@ -769,7 +762,7 @@ public final class Project { // swiftlint:disable:this type_body_length
 			.zip(with: submodulesSignal)
 			.flatMap(.merge) { dependencies, submodulesByPath -> SignalProducer<(), CarthageError> in
 				return SignalProducer<(Dependency, PinnedVersion), CarthageError>(dependencies)
-					.map { (dependency, version) -> SignalProducer<(), CarthageError> in
+					.flatMap(.concurrent(limit: 4)) { (dependency, version) -> SignalProducer<(), CarthageError> in
```

Comment:
Hmm, I've missed that comment. It seems the work is eventually piped to a concurrent queue too, in `cloneOrFetch`.